backend/editor/entries.py

```diff
@@ -247,33 +248,45 @@ def full_text_search(text):
     """
     Helper function used for searching a taxonomy
     """
-    normalized_text = re.sub(r"([^A-Za-z0-9])", r"\\\1", text) # Escape special characters
+    # Escape special characters
+    normalized_text = re.sub(r"([^A-Za-z0-9 _])", r"\\\1", text)
+    normalized_id_text = re.sub(r"([^A-Za-z0-9 _])", r"\\\1", normalizing(text))
+
     text_query_exact = "*" + normalized_text + '*'
     text_query_fuzzy = normalized_text + "~"
-
-    # Fuzzy search on two indexes
+    text_id_query_exact = normalized_id_text + "~"
+    text_id_query_fuzzy = "*" + normalized_id_text + "*"
+    params = {
+        "text_query_fuzzy" : text_query_fuzzy,
+        "text_query_exact" : text_query_exact,
+        "text_id_query_fuzzy" : text_id_query_fuzzy,
+        "text_id_query_exact" : text_id_query_exact 
+    }
+
+    # Fuzzy search and "* search" search on two indexes
     # Fuzzy search has more priority, since it matches more close strings
-    query = f"""
-        CALL db.index.fulltext.queryNodes("nodeSearchIds", $textqueryfuzzy) YIELD node, score
-        WHERE score > 0.2
-        RETURN node.id
-        UNION
-        CALL db.index.fulltext.queryNodes("nodeSearchTags", $textqueryfuzzy) YIELD node, score
-        WHERE score > 0.2
-        RETURN node.id
-    """
-    result = [record["node.id"] for record in session.run(query, {"textqueryfuzzy" : text_query_fuzzy})]
-
-    # If result is empty, search with * symbol on the two indexes to get exact matches
-    if (not result):
-        query = f"""
-            CALL db.index.fulltext.queryNodes("nodeSearchIds", $textqueryexact) YIELD node, score
-            WHERE score > 0.2
-            RETURN node.id
+    query = """
+        CALL {
+                CALL db.index.fulltext.queryNodes("nodeSearchIds", $text_id_query_fuzzy)
+                yield node, score as score_
+                return node, score_ * 3 as score
             UNION
-            CALL db.index.fulltext.queryNodes("nodeSearchTags", $textqueryexact) YIELD node, score
-            WHERE score > 0.2
-            RETURN node.id
-        """
-        result = [record["node.id"] for record in session.run(query, {"textqueryexact" : text_query_exact})]
+                CALL db.index.fulltext.queryNodes("nodeSearchTags",  $text_query_fuzzy)
+                yield node, score as score_
+                return node, score_ * 5 as score
+            UNION
+                CALL db.index.fulltext.queryNodes("nodeSearchIds",  $text_id_query_exact)
+                yield node, score as score_
+                return node, score_ as score
+            UNION
+                CALL db.index.fulltext.queryNodes("nodeSearchTags", $text_query_exact)
+                yield node, score as score_
+                return node, score_ as score 
+        }
+        with node.id as node, score
+        RETURN node, avg(score) as score

```

Comment:
It's very strange to use an average instead of a sum here. 
It means if it matches both fuzzy and exact it may have a lower score ?

Eventually you want to use "max" ?

Reply:
@alexgarel I'm not sure I'd want to use "max", since * searches can have more score over fuzzy searches. But, fuzzy searches would have more accuracy in terms of finding the closest match. So, an average of all the scores must be taken to figure out the best possible match.

Reply:
Atleast in my tests, "max" might give the wrong closest node, and "avg" will find the best possible match. @alexgarel how do you think I should proceed?